pytensor/tensor/subtensor.py

```diff
@@ -2120,16 +2120,12 @@ def make_node(self, x, ilist):
         out_shape = (ilist_.type.shape[0], *x_.type.shape[1:])
         return Apply(self, [x_, ilist_], [TensorType(dtype=x.dtype, shape=out_shape)()])
 
-    def perform(self, node, inp, out_):
+    def perform(self, node, inp, output_storage):
         x, i = inp
-        (out,) = out_
-        # Copy always implied by numpy advanced indexing semantic.
-        if out[0] is not None and out[0].shape == (len(i),) + x.shape[1:]:
-            o = out[0]
-        else:
-            o = None
 
-        out[0] = x.take(i, axis=0, out=o)
+        # Numpy take is always slower when out is provided
+        # https://github.com/numpy/numpy/issues/28636
```

Comment:
The linked issue suggets we could do `out[...] = take(arr, idx)` for speed

Reply:
Oh I see now that you're handling it with special C code

Reply:
That snippet was a solution for the case of errors to still not corrupt the original array, which is not what we are concerned about